app/forms/pet_form.rb

```diff
@@ -1,25 +1,18 @@
 class PetForm
   include ActiveModel::Model
 
-  attr_accessor :photos, :pet_id
+  attr_accessor :photos
 
-  # def initialize(attributes = nil, post: PetImage.new)
-  #   attributes ||= default_attributes
-  #   super(attributes)
-  # end
+  def initialize(pet_id: nil, pet_images: [])
+    @photos = []
+    pet_images.reject(&:blank?)&.each do |image|
+      @photos << PetImage.new(photo: image, pet_id: pet_id)
+    end
```

Comment:
ワンライナーでいきたいなら↓のような書き方でも OK です。
```suggestion
    @photos = pet_images.reject(&:blank?)&.map { |image| PetImage.new(photo: image, pet_id: pet_id) }
```

Reply:
ちなみに空配列入る現象の対策については以下が参考になりそうです。
https://qiita.com/keita44_f4/items/fd899beb1efdf6f1d648